R/tbl_split.R

```diff
@@ -1,66 +1,263 @@
+#'   definitions `tbl_split_by_rows` and `tbl_split_by_columns`.
+#'
+#' @details
+#' Run `gtsummary::show_header_names()` to print all column names to split by.
 #'
 #' @examples
-#' tbl <-
-#'   tbl_summary(trial) |>
-#'   tbl_split(variables = c(marker, grade))
+#' # create standard table
+#' tbl <- tbl_summary(trial, by = trt)
+#'
+#' # split by rows ---------------------------
+#' tbl_list_rows <- tbl |>
+#'   tbl_split_by_rows(variables = c(marker, grade))
+#'
+#' # print column names
+#' gtsummary::show_header_names(tbl)
+#'
+#' # split by columns ------------------------
+#' tbl_list_cols <- tbl |>
+#'   tbl_split_by_columns(groups = list("stat_1", "stat_2"))
+#'
+#' # take the list of splitted rows and split again by columns
+#' tbl_list_rows_cols <- tbl_list_rows |>
+#'   tbl_split_by_columns(groups = list("stat_1", "stat_2"))
+#'
+#' # Handle footnotes -------------------------
+#' tbl_fc <- trial |>
+#' tbl_summary(by = trt, missing = "no") |>
+#'   modify_footnote_header(
+#'     footnote = "All but four subjects received both treatments in a crossover design",
+#'     columns = all_stat_cols(),
+#'     replace = FALSE
+#'   ) |>
+#'   modify_footnote_body(
+#'     footnote = "Tumor grade was assessed _before_ treatment began",
+#'     columns = "label",
+#'     rows = variable == "grade" & row_type == "label"
+#'   ) |>
+#'   modify_spanning_header(
+#'     c(stat_1, stat_2) ~ "**TRT**"
+#'   ) |>
+#'   modify_abbreviation("I = 1, II = 2, III = 3") |>
+#'   modify_caption("_Some caption_") |>
+#'   modify_footnote_spanning_header(
+#'     footnote = "Treatment",
+#'     columns = c(stat_1)
+#'   ) |>
+#'   modify_source_note("Some source note!")
+#'
+#' # lets split it!
+#' tbl_list <- tbl_fc |>
+#'   tbl_split_by_rows(variables = c(marker, grade), footnotes = "last", caption = "first")
+#'
+#' @name tbl_split_by
 NULL
 
 #' @export
-#' @rdname tbl_split
-tbl_split <- function(x, ...) {
-  UseMethod("tbl_split")
+#' @rdname tbl_split_by
+tbl_split_by_rows <- function(x, variables, footnotes = "each", caption = "each") {
+  set_cli_abort_call()
+
+  # list map -------------------------------------------------------------------
+  if (inherits(x, "list")) {
+    tbl_list <- map(
+      .x = x,
+      .f = tbl_split_by_rows,
+      variables = {{ variables }}
+    )
+  } else {
+    # check inputs ---------------------------------------------------------------
+    check_class(x, "gtsummary")
+
+    # process inputs -------------------------------------------------------------
+    cards::process_selectors(
+      data = scope_table_body(x$table_body),
+      variables = {{ variables }}
+    )
+    # adding last variable
+    variables <- variables |> union(dplyr::last(x$table_body$variable))
+
+    footnotes <- arg_match(footnotes, values = c("each", "first", "last"))
+    # if (missing(footnotes)) {
+    #   footnotes <- get_theme_element("tbl_summary-arg:footnotes", default = translate_string(footnotes)) # styler: off
+    # }
```

Comment:
Delete this?

Reply:
yes! I kept it just because I wondered if we want some theme input on this

Reply:
For themes, I generally don't add the theme elements unless I need them.  I went a bit overboard previously, and prefer to maintain as few of them as possible.

Reply:
understood! Cleaned up